### PR TITLE
Create create_recipe_template.mdx

### PR DESCRIPTION
**Pull request by rgreenberg1**

### CODE REVIEW

File: docs/create_recipe_template.mdx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+metaDescription: "Creating Sparsification Recipes with SparseML"
+index: 1000 
+---
+
+# Creating Sparsification Recipes with SparseML Recipe Template
+
+This page explains how to create recipes using the SparseML Recipe Template local API. 
+
+The SparseML Recipe Template API is a simple API that outtakes in an optional path to a dense Pytorch model as input and returns a [Sparsification recipe](https://docs.neuralmagic.com/user-guide/recipes) for that given model in the form of a `recipe.md` file. If no model is supplied, SparseML Recipe Template will generate a default Sparsification Recipe for a generic model. The Sparsification recipe that is outputted by a call to the SopaseML Recipe Template API can then be applied to a model in [Pytorch, Keras, or TensorFlow V1 training pipelines](https://docs.neuralmagic.com/user-guide/recipes/enabling) to train asparse model, all with just a few lines of code. 
+
```
Comment (rahul-tuli) [inline docs/create_recipe_template.mdx:13]:
```suggestion
SparseML Recipe Template is a simple API to generate sparsification recipes on the fly. In the most basic use case it takes in information about the pruning and quantization techniques and produces a relevant recipe.
Optionally path to a `torch.jit` loadable model can also be specified in which case the recipe produced is specific to the architecture of the specified model.  The  sparsification recipes are produced in `markdown` or `yaml` format which can be directly used across all SparseML utilities. Learn more about sparsification recipes [here](https://docs.neuralmagic.com/user-guide/recipes)
```

The reason being we can also specify a sparse model in recipe template, and the code as of now does not support application to Tensorflow, or Keras models

File: docs/create_recipe_template.mdx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+SparseML Recipe Template is built for users who may be inclined to create their recipes by hand to enable more
+fine-grained control or add custom modifiers when sparsifying a new model from scratch with a single API command. 
+
+
+## Overview
+
+SparseML Recipe Template takes in the path to an optional Pytorch model that the you wish to generate a Sparsification recipe for; if none is provided, Recipe Template will generate a generic recipe based on the defaults and designated parameters provided. You may supply a path to a torchscript model (ie *.pt) that can be loaded with `torch.jit.load`. If you are using the python API, you may also supply an already constructed pytorch Module.
```
Comment (rahul-tuli) [inline docs/create_recipe_template.mdx:20]:
```suggestion
SparseML Recipe Template can optionally take in the path to an `torch.jit` loadable model that a user wishes to generate a Sparsification recipe for; if no such model is provided, a generic recipe based on good defaults and designated parameters is produced.  If you are using the python API, you may also supply an already instantiated PyTorch Module.
```

File: docs/create_recipe_template.mdx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+
+You have many different options for configuring the Recipe Template API to generate a sparse model to your choosing leveraging techniques like pruning and quantization. Recipe Template has many available arguements which allows you to fully customize the recipe that gets created. A handful of them are required to help designate the techniques used in generating a Sparsification recipe.
```
Comment (rahul-tuli) [inline docs/create_recipe_template.mdx:23]:
```suggestion
This utility also provides different options for configuring the generated recipe according to required sparsification techniques. Users are encouraged to change the recipes according to their specific needs.
```

File: docs/create_recipe_template.mdx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+To run the SparseML Recipe Template API, you use the following command:
```
Comment (rahul-tuli) [inline docs/create_recipe_template.mdx:25]:
```suggestion
To learn more about the usage invoke the Recipe Template cli with `--help` option: 
```

File: docs/create_recipe_template.mdx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+SparseML Recipe Template takes in the path to an optional Pytorch model that the you wish to generate a Sparsification recipe for; if none is provided, Recipe Template will generate a generic recipe based on the defaults and designated parameters provided. You may supply a path to a torchscript model (ie *.pt) that can be loaded with `torch.jit.load`. If you are using the python API, you may also supply an already constructed pytorch Module.
+
+
+You have many different options for configuring the Recipe Template API to generate a sparse model to your choosing leveraging techniques like pruning and quantization. Recipe Template has many available arguements which allows you to fully customize the recipe that gets created. A handful of them are required to help designate the techniques used in generating a Sparsification recipe.
+
+To run the SparseML Recipe Template API, you use the following command: 
+
+```bash
+  SparseML.recipe_template(model[optional]) --args[optional]
```
Comment (rahul-tuli) [inline docs/create_recipe_template.mdx:28]:
```suggestion
  sparseml.recipe_template --help
```

File: docs/create_recipe_template.mdx [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+```
+
+Arguments can be presented to designate different aspects of the recipe such as whether it should include pruning and if so, what algorithm should be used, quantization and if so, what algorithm should be used, as well as the learning rate type.
```
Comment (rahul-tuli) [inline docs/create_recipe_template.mdx:31]:
```suggestion
The arguments specified configure different aspects of the recipe such as whether it should include pruning and if so, what algorithm should be used, quantization and if so, what algorithm should be used, as well as the learning rate schedule function type to mention a few. 
```

File: docs/create_recipe_template.mdx [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+```bash
+--pruning true|false(default)|ALGORITHM
+ALGORITHM: global magnitude pruning, gradual magnitude pruning – sparsity per layer, ACDC, Movement, MFAC, OBC
+--quantization true|false(default)|TARGET_HARDWARE(default to QAT algorithm ex: vnni, tensorrt)
+--lr TYPE
+TYPE: constant, stepped, cyclic, exponential, linear
+```
+
+### Creating the Recipe Template
+Once SparseML.recipe_template is run, a base recipe template will be created with the necessary top level variables based on the compression methods specified (ie pruning, quantization). The repo will maintain versions for pruning, qat, and pruning+qat.  These templates will either be python strings with `{variable}` placeholders left in for on the fly formatting, or functions that take in the desired variables and return the constructed template.
```
Comment (rahul-tuli) [inline docs/create_recipe_template.mdx:43]:
```suggestion
Once `sparseml.recipe_template` is run, a base recipe template will be created with the necessary top level variables based on the compression methods specified (ie pruning, quantization). The repo will maintain versions for pruning, qat, and pruning+qat.  These templates will either be python strings with `{variable}` placeholders left in for on the fly formatting, or functions that take in the desired variables and return the constructed template.
```

not too sure what do you mean by maintaining versions

File: docs/create_recipe_template.mdx [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+- Convenience functions to replace manager from_yaml (ie manager.sparsification(...))
+- For LR modifiers:
+  - When pruning, cycle through init_lr and final_lr
+  - When finetuning, cycle through init_lr and final_lr
+  - When quantizing, hold constant at final_lr
+
+
+### Additional Recipe Template Modifyers
+
+In addition to the base arguments listed above, you can also fine tune specific parameters at a more detialed level such as Epoch, LR, Pruning, QAT, and Modifyers to guide the recipe you want created for your model. To view the exaustive list of arguments available, run the following help command:
```
Comment (rahul-tuli) [inline docs/create_recipe_template.mdx:75]:
```suggestion
When using the python api, you can also fine tune specific parameters at a more detialed level such as Epoch, LR, Pruning, QAT, and Modifiers in addition to the base arguments to configure the generated recipe based on your needs.
```

CLI doesn't have these arguments they are only present in the python api

File: docs/create_recipe_template.mdx [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+
+```bash
+  SparseML.recipe_template() --help
+```
```
Comment (rahul-tuli) [inline docs/create_recipe_template.mdx:79]:
```suggestion
```

**Review by rahul-tuli — COMMENTED** [review]
Some preliminary comments, one thing to note is we should try to avoid using too many pronouns such as you and your in a professional document.

Great first pass though

**Review by rahul-tuli — COMMENTED** [review]
Maybe @jeanniefinks is a better person to go over the grammar

File: docs/create_recipe_template.mdx [hunk 10]
```diff
@@ -0,0 +1,86 @@
+---
+title: "Creating with SparseML Recipe Template"
```
[the remark this answers is not in the document]
Comment (rgreenberg1) [reply docs/create_recipe_template.mdx:2]: I actually agree with you on the you/your. I like to say users. I looked at some of our other docs content and it seems like we use the you like here: https://docs.neuralmagic.com/user-guide/recipes/creating. Something I will discuss with the team though on deciding using it or not.

File: docs/create_recipe_template.mdx (hunk 7 above)
Comment (rgreenberg1) [reply docs/create_recipe_template.mdx:43]: I will just get rid of that sentence, if confusing. I took that from the PRD, if it makes no sense let's scrap it.

**Comment by rahul-tuli** [comment]
Great first draft!

**Pull request merged**